server/src/test/java/keywhiz/service/resources/BatchSecretDeliveryResourceTest.java

```diff
@@ -0,0 +1,163 @@
+import static org.mockito.Mockito.when;
+
+public class BatchSecretDeliveryResourceTest {
+    private static final ApiDate NOW = ApiDate.now();
+
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+
+    @Mock
+    SecretController secretController;
+    @Mock
+    AclDAO aclDAO;
+    @Mock
+    ClientDAO clientDAO;
+    BatchSecretDeliveryResource batchSecretDeliveryResource;
+
+    final Client client = new Client(0, "principal", null, null, null, null, null, null, null, null, false,
+            false);
+
+    final Secret secret = new Secret(0, "secret_name", null, () -> "secret_value", "checksum", NOW, null, NOW, null,
+            null, null, null, 0, 1L, NOW, null);
+    final Secret secretBase64 = new Secret(1, "Base64With=", null, () -> "SGVsbG8=", "checksum", NOW, null, NOW,
+            null, null, null, null, 0, 1L, NOW, null);
+
+    final Secret secret2 = new Secret(2, "secret2_name", null, () -> "secret_value2", "checksum2", NOW, null, NOW, null,
+            null, null, null, 0, 1L, NOW, null);
+
+
+    @Before
+    public void setUp() {
+        batchSecretDeliveryResource = new BatchSecretDeliveryResource(secretController, aclDAO, clientDAO);
+    }
+
+    @Test
+    public void returnsSingleSecretWhenAllowed() throws Exception {
+        SanitizedSecret sanitizedSecret = SanitizedSecret.fromSecret(secret);
+
+        ImmutableList<String> secretnames = ImmutableList.of(sanitizedSecret.name());
+        BatchSecretRequest req = BatchSecretRequest.create(secretnames);
+
+        when(aclDAO.getBatchSanitizedSecretsFor(client, secretnames))
+                .thenReturn(List.of(sanitizedSecret));
+        when(secretController.getSecretsByName(secretnames))
+                .thenReturn(List.of(secret));
+
+        List<SecretDeliveryResponse> response = batchSecretDeliveryResource.getBatchSecret(client, req);
+        assertThat(response).isEqualTo(ImmutableList.of(SecretDeliveryResponse.fromSecret(secret)));
+    }
+
+    @Test
+    public void returnsMultipleSecretsWhenAllowed() throws Exception {
+        SanitizedSecret sanitizedSecret = SanitizedSecret.fromSecret(secret);
+        SanitizedSecret sanitizedSecret2 = SanitizedSecret.fromSecret(secret2);
+
+        ImmutableList<String> secretnames = ImmutableList.of(sanitizedSecret.name(), sanitizedSecret2.name());
+        BatchSecretRequest req = BatchSecretRequest.create(secretnames);
+
+        when(aclDAO.getBatchSanitizedSecretsFor(client, secretnames))
+                .thenReturn(List.of(sanitizedSecret, sanitizedSecret2));
+        when(secretController.getSecretsByName(secretnames))
+                .thenReturn(List.of(secret, secret2));
+
+        List<SecretDeliveryResponse> response = batchSecretDeliveryResource.getBatchSecret(client, req);
+
+        // The order doesn't matter
+        boolean matches = false;
+
+        if (response.equals(ImmutableList.of(SecretDeliveryResponse.fromSecret(secret), SecretDeliveryResponse.fromSecret(secret2)))) {
```

Comment:
`assertThat(list).containsExactlyInAnyOrder(item1, item2, item3...)` can simplify this